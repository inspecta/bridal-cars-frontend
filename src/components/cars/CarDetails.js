import React from 'react';
import { useNavigate, useLocation, Link } from 'react-router-dom';
import { FaAngleLeft } from 'react-icons/fa';

const CarDetails = () => {
  const navigate = useNavigate();

  const { state } = useLocation();
  const selectedCar = state.cars;
  // const user = localStorage.getItem('user');

  if (!selectedCar) {
    return <div>Car not found.</div>;
  }

  const handleClick = () => {
    navigate('/');
  };

  // const handleReserve = () => {
  //   navigate('/reserve-car');
  // };

  return (
    <>
      <div className="car-details">
        <div className="car-details-photo">
          <img
            src={selectedCar.photo}
            style={{ width: '100%' }}
            alt="Bridal car"
          />
        </div>
        <div className="car-details-info">
          <h2>{selectedCar.name}</h2>
          <div className="car-details-more-info">
            <p>{selectedCar.description}</p>
            <p>
              Reservation Price:
              {selectedCar.price}
            </p>
            <p>
              Car Type:
              {selectedCar.model}
            </p>
            <p>
              Created Date:
              {selectedCar.created_at}
            </p>
          </div>
          <div className="add-car-btn">
            <Link
              to="/reservation-form"
              state={{
                car: selectedCar,
              }}
            >
              Reserve
            </Link>
          </div>
        </div>
      </div>
      <div style={{ textAlign: 'center' }}>
        <button onClick={() => handleClick()} type="button">
          <FaAngleLeft />
        </button>
      </div>
    </>
  );
};

export default CarDetails;
