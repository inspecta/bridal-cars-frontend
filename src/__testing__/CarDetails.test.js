import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import CarDetails from "../components/cars/CarDetails";
import { Provider } from "react-redux";

describe("CarDetails", () => {
  const selectedCar = {
    name: "Mercedes",
    photo: "mercedes_photo.jpg",
    description: "A very comfortable car",
    price: 12,
    model: "W-14",
    reserved: false,
    created_at: '09/03/2022',
  };

  it("renders the car details component", () => {
    const location = {
      state: {
        cars: selectedCar
      },
    };

    const { asFragment } = render(
      <BrowserRouter>
        <CarDetails location={location} />
      </BrowserRouter>,
    );

    expect(asFragment).toMatchSnapshot();
  })
})