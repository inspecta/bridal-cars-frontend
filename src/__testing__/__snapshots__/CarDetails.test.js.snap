// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarDetails renders the car details component 1`] = `[Function]`;
